app/services/datastream_importer.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+# This class imports pre-parsed datastream info into the compliance DB
+class DatastreamImporter
+  def initialize(datastream_filenames)
+    @datastreams = datastream_filenames.map do |datastream_filename|
+      OpenscapParser::Datastream.new(File.read(datastream_filename))
+    end
+  end
+
+  def import!
+    @datastreams.map do |ds|
+      Xccdf::Benchmark.transaction do
+        benchmark = save_benchmark(benchmark: ds.benchmark)
+        profiles = save_profiles(benchmark: benchmark, profiles: ds.profiles)
+        # save_rules(ds.rules)
+      end
+    end
+  end
+
+  private
+
+  def save_benchmark(benchmark: OpenscapParser::Benchmark.new)
+    benchmark = Xccdf::Benchmark.find_or_initialize_by(
+      ref_id: benchmark.id,
+      version: benchmark.version
+    )
+
+    benchmark.update(
```

Comment:
Nitpick, why not chaining all like:

```ruby
Xccdf::Benchmark.find_or_initialize_by(
      ref_id: benchmark.id,
      version: benchmark.version
).update(
     title: benchmark.title,
     description: benchmark.description
)
```

The current code is not updating the benchmark as far as I can see, as the argument `benchmark` is overshadowed by `benchmark = Xccdf::Benchmark.find_or...`

Reply:
Good catch about the overshadowing. I wanted this method to take the OpenscapParser benchmark and return an ActiveRecord benchmark because we need the AR benchmark ID to save the rules and profiles later. The `update()` method returns true or false - that's why I've created this intermediate variable.